crypto/algapi.c

```diff
@@ -1028,6 +1028,61 @@ void __crypto_xor(u8 *dst, const u8 *src1, const u8 *src2, unsigned int len)
 }
 EXPORT_SYMBOL_GPL(__crypto_xor);
 
+// Computes dst = src1 ^ strrev(src2)
+void __crypto_xor_reverse(u8 *dst, const u8 *src1, const u8 *src2, unsigned int len)
```

Comment:
Modify this code to only perform the byteswap and move it to `polyval.c`. Calling the byteswap, then `crypto_xor` would have the same effect and would be cleaner.